docs/DesignDocs/SGX_QuoteEx_Integration.md

```diff
@@ -165,79 +167,66 @@ on the same platform.
-of the installation of the two libraries.
 
-When both libraries and the background services are all installed and configured,
-the quote-ex should library takes precedence, as it supports more evidence
-formats.
+* When only the DCAP library is installed, it will be used.
+* When only the quote-ex library is installed, it will be used.
+* When both libraries are installed:
+    * If the DCAP library is configured for in-process quote generation, the DCAP
+    library is chosen, so this default behavior for existing applications is
+    preserved.
+    * Otherwise if the DCAP library is configured for out-of-process quote
+    generation, the quote-ex library will be used, as it supports more evidence
+    formats.
 
 #### Options for Host-side Plugin Library Link with the SGX DCAP and quote-ex Libraries
 
 There are several options for the OE SDK host-side plugin library to
 link with the SGX DCAP and quote-ex libraries.
-Experiments in implementation will help choose the most suitable option.
-In any case, from the software stack point of view,
-the choice only impacts the implementation of the host-side plugin library.
-It does not impact the enclave-side plugin library or the OCALL interface.
+From the software stack point of view,
+the options only differ in the implementation of the host-side plugin library.
+They share the same enclave-side plugin library implementation and the same
+OCALL interface.
 
-##### Option 1: Dynamic Detection and Loading of the Two Libraries
+##### Option 1: Runtime Detection and Loading of the Two Libraries
 
 With this option, the OE SDK host-side plugin library dynamically detects
-the presence of the two libraries and loads them at runtime.
-If the quote-ex library is present, it loads this library and calls into it
-to check if the dependent background service is available.
-If so the quote-ex library is used. Otherwise the DCAP library is loaded
-and used.
+the presence of the two libraries, and choose to use one of them in the
+priority defined in the previous section.
 
 ##### Option 2: Built-time Link with the quote-ex Library
 
-Run-time dynamic detection and loading of multiple shared libraries
-complicates implementation.
-It also increases the risk of API version mismatch between OE SDK and
-the loaded libraries, since the API version of the loaded libraries is not
-checked at build time against library headers.
-
 As described previously, the quote-ex library supports a superset of formats
 as compared to the DCAP library, though it always depends on
 a background service for quote generation.
 
 If on SGX platforms the OE SDK always installs with the AESM background service
 (as a hard dependency), then it is possible for the host-side plugin library
 to be linked at build-time only with the quote-ex library. With this option,
-the dependency on the DCAP library is dropped.
+the dependency on the DCAP library can be dropped.
```

Comment:
Just for my understanding, is there any reason to soften this statement? The previous version asserts that we would stop supporting in-proc QE/PCE loading i.e. removing DCAP is a _feature_ of this option. It technically doesn't matter since we're not taking this option, but reading it document as a whole, I think it muddles the distinction between the options unnecessarily and the previous version is clearer in intent.

Reply:
I think what I wanted to mean is something like "the dependency on the DCAP library will be dropped", which would happen in the future if and when this option is taken.

Reply:
Yeah, I think you should state that as "is/will be" for this option then. I'll approve with the expectation that this is fixed up.

Reply:
Thanks! This is fixed in the latest commit.